src/praisonai-agents/praisonaiagents/agent/agent.py

```
@@ -551,6 +551,7 @@ def __init__(
         skills: Optional[Union[List[str], str, Dict[str, Any], 'SkillsConfig']] = None,
         approval: Optional[Union[bool, str, Dict[str, Any], 'ApprovalConfig', 'ApprovalProtocol']] = None,
         tool_timeout: Optional[int] = None,  # P8/G11: Timeout in seconds for each tool call
+        tool_retry_policy: Optional['RetryPolicy'] = None,  # Retry policy for tool execution with exponential backoff  
```

Ruff (0.15.15)

[error] Undefined name `RetryPolicy`

(F821)